### PR TITLE
Adding a weather microservice

### PR DESCRIPTION
**Pull request by saijel**
- created a weather microservice for port 8088
- added classification data
- created a testing thrift client
- created Kubernetes .yaml scripts

### CODE REVIEW

**Comment by k0105** [comment]
So I can finally connect a face with the name. A pleasure to make your acquaintance. Will we be cooperating or is development going to stay an isolated thing?

File: lucida/weather/server/WeatherServer.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from thrift.transport import TTransport
+from thrift.protocol import TBinaryProtocol
+from thrift.server import TServer
+
+import urllib2
+import json
+import urllib
+
+port = 8088
+owm_api_key = "362537b891e6df03a316e82565fe4df3"
```
Comment (jhauswald) [inline lucida/weather/server/WeatherServer.py]: ew Magic Numbers, can you move this to a config and put in the README that it needs to be set or something? Should be a "secret"

File: lucida/weather/server/WeatherServer.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     - LUCID: ID of Lucida user
+     - query: query
+    """
+    def infer(self, LUCID, query):
+        result = "No weather found for %s" % query.content[0].data[0]
+        url_location = urllib.quote_plus(query.content[0].data[0])
+
+        try:
+            # weather from Weather Underground
+            f = urllib2.urlopen('http://api.wunderground.com/api/%s/conditions/q/%s.json' % (wu_api_key, url_location) )
```
Comment (jhauswald) [inline lucida/weather/server/WeatherServer.py]: should be a gloval

File: lucida/weather/server/WeatherServer.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            parsed_json = json.loads(json_string)
+            if 'error' not in parsed_json['response'] and 'current_observation' in parsed_json:
+                weather = parsed_json['current_observation']['weather']
+                temp = parsed_json['current_observation']['temperature_string']
+                city = parsed_json['current_observation']['display_location']['full']
+                result = "Current weather in %s is %s %s" % (city, weather, temp)
+                print "From Weather Underground: %s" % result
+            else:
+                # weather from Open Weather Map
+                f = urllib2.urlopen('http://api.openweathermap.org/data/2.5/weather?q=%s&appid=%s&units=imperial&type=like' % (url_location, owm_api_key))
```
Comment (jhauswald) [inline lucida/weather/server/WeatherServer.py]: global or some config

File: lucida/weather/server/WeatherServer.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+from thrift.transport import TSocket
+from thrift.transport import TTransport
+from thrift.protocol import TBinaryProtocol
+from thrift.server import TServer
+
+import urllib2
+import json
+import urllib
+
+port = 8088
```
Comment (jhauswald) [inline lucida/weather/server/WeatherServer.py]: we really need to rethink how all the ports are set, it's super messy to have them nested deep into the code. then you need to worry about setting them correctly within the CMD center as well

**Comment by jhauswald** [comment]
This generally looks good, I need to test it out but I like the content.

File: lucida/weather/server/WeatherServer.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    Do nothing
+    Parameters:
+     - LUCID: ID of Lucida user
+     - spec: spec
+    """
+    def create(self, LUCID, spec):
+        return
+
+    """
```
Comment (jhauswald) [inline lucida/weather/server/WeatherServer.py]: these docstrings should be inside the function header, check how we do it internally

**Comment by jhauswald** [comment]
We should probably merge #163 before this right? I'd want to be able to test this with your tmux script (since that makes it actually easier to test :P)

**Comment by saijel** [comment]
Yes, I completely agree. Let me work on the changes for the tmux PR first
and then add changes based on that merge within this PR.

---
Saijel Mokashi
https://clinc.com

**Comment by k0105** [comment]
Why is this code mixing urllib2 and urllib? And do you want me to keep commenting or shall I keep to myself?

**Pull request closed without merging**